eclipse-collections/src/main/java/org/eclipse/collections/impl/list/mutable/FastList.java

```diff
@@ -803,11 +799,7 @@ public MutableByteList collectByte(ByteFunction<? super T> byteFunction)
     @Override
     public <R extends MutableByteCollection> R collectByte(ByteFunction<? super T> byteFunction, R target)
     {
-        for (int i = 0; i < this.size; i++)
-        {
-            target.add(byteFunction.byteValueOf(this.items[i]));
-        }
-        return target;
+        return InternalArrayIterate.collectByte(this.items, this.size, byteFunction, target);
```

Comment:
I have left the flatCollectX untouched but it seems reasonable to push them in the internal array iterable as well - i know atm they are only used in one place but seems it will be more consistent for all of these to come from internal array iterate. Further, if in future we do need to reuse this functionality it would already be moved to a common util class. Any thoughts?

Reply:
Makes sense - thx